catroid/src/test/java/org/catrobat/catroid/test/content/actions/SetThreadColorActionTest.kt

```diff
@@ -0,0 +1,128 @@
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.content.actions
+
+import com.badlogic.gdx.graphics.Color
+import com.badlogic.gdx.scenes.scene2d.Action
+import com.badlogic.gdx.scenes.scene2d.actions.Actions
+import com.badlogic.gdx.scenes.scene2d.actions.SequenceAction
+import junit.framework.Assert.assertEquals
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.content.Scope
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.content.actions.SetThreadColorAction
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.test.StaticSingletonInitializer.Companion.initializeStaticSingletonMethods
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class SetThreadColorActionTest {
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        initializeStaticSingletonMethods()
+        sprite = Sprite("testSprite")
+        createTestableSetThreadColorAction(
+            sprite,
+            SequenceAction(), Formula(COLOR_TEST_VALUE_BLACK)
+        )?.act(1.0f)
+    }
+
+    @Test
+    fun testNormalBehaviorColorChange() {
+        assertEquals(Color.BLACK, sprite?.embroideryThreadColor)
+        createTestableSetThreadColorAction(
+            sprite, SequenceAction(),
+            Formula(COLOR_TEST_VALUE_WHITE)
+        )?.act(1.0f)
+        assertEquals(Color.WHITE, sprite?.embroideryThreadColor)
+        createTestableSetThreadColorAction(
+            sprite, SequenceAction(),
+            Formula(COLOR_TEST_VALUE_BLUE)
+        )?.act(1.0f)
+        assertEquals(Color.BLUE, sprite?.embroideryThreadColor)
+    }
+
+    @Test(expected = NullPointerException::class)
+    fun testNullSprite() {
+        val action = sprite?.actionFactory?.createSetThreadColorAction(
+            null,
+            SequenceAction(), Formula(COLOR_TEST_VALUE_BLACK)
+        )
+        action?.act(1.0f)
+    }
+
+    @Test
+    fun testNullFormula() {
+        createTestableSetThreadColorAction(sprite, SequenceAction(), null)?.act(1.0f)
+        assertEquals(COLOR_INITIALIZED_VALUE, sprite?.embroideryThreadColor)
+    }
+
+    @Test(expected = StringIndexOutOfBoundsException::class)
```

Comment:
See comment in `SetThreadColorAction`: This test should not expect an exception but rather fail gracefully and leave the thread color unchanged.